packages/swap/src/routes/v2/swap.ts

```diff
@@ -149,32 +148,36 @@ router.get(
       failedSwap?.depositTransactionRef ??
       undefined;
 
-    const sortedSwaps = swapRequest?.swaps.filter(isEgressableSwap).sort((a, b) => {
-      if (a.swapExecutedAt && b.swapExecutedAt) {
-        return a.swapExecutedAt.valueOf() - b.swapExecutedAt.valueOf() ? 1 : -1;
-      }
-      return a.swapExecutedAt && !b.swapExecutedAt ? 1 : 0;
-    });
+    let originalInputAmount;
+
+    if (swapRequest) {
+      originalInputAmount = swapRequest.fees.reduce(
+        (acc, fee) => (fee.asset === swapRequest.srcAsset ? acc.minus(fee.amount) : acc),
```

Comment:
might need to filter to only subtract ingress fee

Reply:
for older swaps the broker fee too